mongodbatlas/data_source_mongodbatlas_federated_query_limits.go

```diff
@@ -0,0 +1,80 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/id"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func dataSourceMongoDBAtlasFederatedDatabaseQueryLimits() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMongoDBAtlasFederatedDatabaseQueryLimitsRead,
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tenant_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"results": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: schemaMongoDBAtlasFederatedDatabaseQueryLimitDataSource(),
+				},
+			},
+		},
+	}
+}
+
+func dataSourceMongoDBAtlasFederatedDatabaseQueryLimitsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*MongoDBClient).Atlas
+
+	projectID := d.Get("project_id").(string)
+	tenantName := d.Get("tenant_name").(string)
+
+	queryLimits, _, err := conn.DataFederation.ListQueryLimits(ctx, projectID, tenantName)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error getting federated database query limits for project (%s), tenant (%s), error: %s", projectID, tenantName, err))
+	}
+
+	if results := flattenFederatedDatabaseQueryLimits(projectID, tenantName, queryLimits); results != nil {
+		if err := d.Set("results", results); err != nil {
+			return diag.FromErr(fmt.Errorf(errorFederatedDatabaseQueryLimit, "results", projectID, err))
+		}
+	}
+
+	d.SetId(id.UniqueId())
+
+	return nil
+}
+
+func flattenFederatedDatabaseQueryLimits(projectID, tenantName string, queryLimits []*matlas.DataFederationQueryLimit) []map[string]interface{} {
+	var federatedDatabaseQueryLimitMap []map[string]interface{}
+
+	if len(queryLimits) > 0 {
```

Comment:
[nit] I like to avoid nested `if` when possible
```suggestion
	if len(queryLimits) == 0 {
	 return federatedDatabaseQueryLimitMap
	}
```